### PR TITLE
Adjusted incorrect Policy CRD verification

### PR DESCRIPTION
**Pull request by fgiorgetti**

### CODE REVIEW

**Review by nluaces — APPROVED** [review]

File: client/policy_validator.go [hunk 1]
```diff
@@ -25,13 +24,12 @@ import (
 
 type PolicyValidationResult struct {
 	err             error
+	crdDefined      bool
```
Comment (grs) [inline client/policy_validator.go:27]: Maybe 'enabled' would be a better name for this field?

Comment (fgiorgetti) [reply client/policy_validator.go:27]: Certainly more friendly. I will change it.

File: client/policy_validator.go [hunk 2]
```diff
@@ -61,13 +59,12 @@ type ClusterPolicyValidator struct {
 	cli           *VanClient
 	dc            *discovery.DiscoveryClient
 	skupperPolicy v1alpha1.SkupperClusterPolicyInterface
-	labelRegex    *regexp.Regexp
+	mockEnabled   bool
```
Comment (grs) [inline client/policy_validator.go:62]: Myabe 'disablePolicyDiscovery' would be a nicer name for this?

Comment (fgiorgetti) [reply client/policy_validator.go:62]: +1

File: client/policy_validator.go [hunk 3]
```diff
@@ -171,7 +168,13 @@ func (p *ClusterPolicyValidator) HasPermission() bool {
 }
 
 func (p *ClusterPolicyValidator) CrdDefined() bool {
+	if p.mockEnabled {
+		return true
+	}
 	if p.dc == nil {
+		if p.cli.RestConfig == nil {
+			return false
```
Comment (grs) [inline client/policy_validator.go:176]: Is this the right result? Feels like it is safer to fail the attempted operation if we cannot determine whether or not policy is enabled?

**Review by grs — APPROVED** [review]

File: client/policy_validator.go (hunk 3 above)
Comment (fgiorgetti) [reply client/policy_validator.go:176]: The `RestConfig` attribute is always set, except when using `newMockClient` to create the `*VanClient` instance.
When it happens (mock tests only), we have to consider CRD as absent, so existing unit tests continue to work.

That's why we now have the `disablePolicyDiscovery` flag. If a unit test needs the policy validator to consider
CRD as present, we set it to true.

**Pull request merged**